### PR TITLE
File contents hashing & decoding fixes, plus new unit tests

### PR DESCRIPTION
**Pull request by EasyRhin0**

### CODE REVIEW

File: src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs [hunk 1]
```diff
@@ -163,38 +163,30 @@ private void EnsureHashExists(FileData file)
             var hasSha256Hash = file.Hashes.Any(x => x.Algorithm == AlgorithmKind.Sha256);
             if (!hasSha256Hash)
             {
-                string content = null;
-
-                if (!string.IsNullOrWhiteSpace(file.Contents.Binary))
+                byte[] data = null;
+                if (file?.Contents.Binary != null)
```
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs]: I did this wrong in the sample I sent you. It needs to be `file.Contents?.Binary` and `file.Contents?.Text`.

File: src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifFileWithContentsTests.cs [hunk 2]
```diff
@@ -16,9 +16,15 @@ namespace Microsoft.Sarif.Viewer.VisualStudio.UnitTests
     [Collection("SarifObjectTests")]
     public class SarifFileWithContentsTests
     {
-        private const string Key = "/item.cpp#fragment";
-        private const string Key2 = "/item2.cpp";
-        private const string Contents = "This is a test file.";
+        private const string key1 = "/item.cpp#fragment";
```
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifFileWithContentsTests.cs]: `PascalCase` for `const` (_e.g._, `Key1`).

File: src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifFileWithContentsTests.cs [hunk 3]
```diff
@@ -97,35 +181,79 @@ public void SarifFileWithContents_SavesContents()
...
+            fileText.Should().Be(expectedContents2);
+        }
+
+        [Fact]
+        public void SarifFileWithContents_DecodesTextContents()
+        {
+            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[key3];
+            var contents = fileDetail.GetContents();
+
+            contents.Should().Be(expectedContents1);
```
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifFileWithContentsTests.cs]: These tests don't verify the hashing code. For each one, verify the file hash value.

File: src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs (hunk 1 above)
Comment (EasyRhin0) [reply src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs]: Fixed

File: src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifFileWithContentsTests.cs (hunk 2 above)
Comment (EasyRhin0) [reply src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifFileWithContentsTests.cs]: Oh right, fixed

File: src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifFileWithContentsTests.cs [hunk 4]
```diff
@@ -97,35 +183,85 @@ public void SarifFileWithContents_SavesContents()
...
+            contents.Should().Be(ExpectedContents2);
+        }
+
+        [Fact]
+        public void SarifFileWithContents_HandlesEmptyBinaryContents()
+        {
+            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[Key5];
+            var contents = fileDetail.GetContents();
+
+            contents.Should().Be(string.Empty);
```
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifFileWithContentsTests.cs]: Add the hash value to show that you don't skip hashing an empty file -- here, and in the next test.

**Pull request merged**